packages/jsapi-utils/src/TableInput.tsx

```diff
@@ -0,0 +1,319 @@
+import React, {
+  useCallback,
+  useState,
+  useRef,
+  useEffect,
+  useMemo,
+  ReactNode,
+} from 'react';
+import classNames from 'classnames';
+import {
+  LoadingOverlay,
+  SearchInput,
+  SelectValueList,
+} from '@deephaven/components';
+import type { LongWrapper, Table } from '@deephaven/jsapi-shim';
+import { PromiseUtils } from '@deephaven/utils';
+import Log from '@deephaven/log';
+import {
+  DecimalColumnFormatterOptions,
+  Formatter,
+  FormatterUtils,
+  FormattingRule,
+  IntegerColumnFormatterOptions,
+} from '.';
+import useTableColumn from './useTableColumn';
+
+import './TableInput.scss';
+
+const log = Log.module('TableInput');
+
+type Value = LongWrapper | string;
+
+interface SelectValueItem {
+  displayValue: string;
+  value: Value;
+  isSelected: boolean;
+}
+
+interface Settings {
+  formatter: FormattingRule[];
+  timeZone: string;
+  defaultDateTimeFormat: string;
+  showTimeZone: boolean;
+  showTSeparator: boolean;
+  defaultDecimalFormatOptions?: DecimalColumnFormatterOptions;
+  defaultIntegerFormatOptions?: IntegerColumnFormatterOptions;
+}
+
+interface TableInputProps {
+  className?: string;
+  columnName: string;
+  settings: Settings;
+  defaultValue: Value[];
+  isInvalid?: boolean;
+  table: Promise<Table>;
+  onChange(items: Value[]): void;
+  onBlur?: () => void;
+}
+
+const SIZE_LIMIT = 250;
+
+const TableInput = (props: TableInputProps): ReactNode => {
```

Comment:
I know I haven't been consistent on this, but for function components and hooks, I prefer to use the `function` declaration like they do in the docs: https://reactjs.org/docs/components-and-props.html
e.g. it should be `function TableInput(props: TableInputProps): ReactNode` instead of `const TableInput = (props: TableInputProps): ReactNode`